### ISSUE TITLE
Dazn Add-on Does not work

### ISSUE THREAD

**Issue by DiegoAntonino** [kodinerds/repo#27]
Hi, 
I have installed Kodi on my raspberry-pi 3 (that runs Rasbian-Stretch),

```
$dpkg --list | grep kodi
ii  kodi                                  2:17.4-1~stretch                     all          Kodi Media Center (arch-independent data package)
ii  kodi-bin                              2:17.4-1~stretch                     armhf        Kodi Media Center (binary data package)
ii  kodi-inputstream-adaptive             2.0.10-1~stretch                     armhf        adaptive inputstream addon for Kodi
ii  kodi-inputstream-adaptive-dbg         2.0.10-1~stretch                     armhf        debug symbols for adaptive inputstream addon for Kodi
ii  kodi-inputstream-rtmp                 1.0.4-1~stretch                      armhf        Rtmp inputstream addon for Kodi
```

As you can see above I have installed "inputstream-adaptive" as well.

First, I wasn't able to install the add-on using the following instructions:

1. Select Install from Zip File > select the repository.kodinerds-6.0.0.zip file you downloaded above and wait for the Add-on enabled notification
2. Select Install from Repository > Kodi Nerds repository > Video Add-ons > DAZN > Install
3. Wait for Add-on enabled notification

I got an error saying:
`20:22:35.150 T:1590686464   ERROR: CAddonInstallJob[plugin.video.dazn]: The dependency on inputstream.adaptive version 2.0.10 could not be satisfied`     

Which is not really true since it's installed :(

So, secondly I try using [Indigo tool](https://www.tvaddons.co/indigo-kodi-krypton/)   
This time add-on was installed, I was able to launch it and log-in to my Dazn account **BUT** when I try to watch any content I got the following error:

```
20:57:32.427 T:1691501312  NOTICE: [plugin.video.dazn] country: ca language: en arguments: {'title': ['beIN Sports (Live Now)'], 'params': ['43s5qho1kdxv7kw3sk36rdz75'], 'id': ['1dje39ox1sc7zzfdw3csa0krl'], 'mode': ['play']}
20:57:33.122 T:1911562240  NOTICE: VideoPlayer: Opening: https://dc1live2cadazn-a.akamaihd.net/out/u/dazn-linear-dc1-002.mpd
20:57:33.123 T:1911562240 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
20:57:33.124 T:1797255936  NOTICE: Creating InputStream
20:57:33.474 T:1797255936   ERROR: AddOnLog: InputStream Adaptive: Could not open / parse mpdURL (https://dc1live2cadazn-a.akamaihd.net/out/u/dazn-linear-dc1-002.mpd)
20:57:33.474 T:1797255936   ERROR: CVideoPlayer::OpenInputStream - error opening [https://dc1live2cadazn-a.akamaihd.net/out/u/dazn-linear-dc1-002.mpd]
20:57:33.474 T:1797255936  NOTICE: CVideoPlayer::OnExit()
20:57:33.474 T:1911562240   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.dazn/?id=1dje39ox1sc7zzfdw3csa0krl&params=43s5qho1kdxv7kw3sk36rdz75&mode=play&title=beIN+Sports+%28Live+Now%29]
20:57:33.475 T:1911562240  NOTICE: CVideoPlayer::CloseFile()
20:57:33.475 T:1911562240  NOTICE: VideoPlayer: waiting for threads to exit
20:57:33.475 T:1911562240  NOTICE: VideoPlayer: finished waiting
20:57:33.476 T:1911562240  NOTICE: CVideoPlayer::CloseFile()
20:57:33.476 T:1911562240  NOTICE: VideoPlayer: waiting for threads to exit
20:57:33.476 T:1911562240  NOTICE: VideoPlayer: finished waiting
```
At this stage I have not clue what I could be wrong so, any help would be more than welcome!

Thanks!
Diego

**Comment by JinRonin** [comment]
First things first:
Please, never install addons from repositories that automatically collect addons from other sources.
Dont install them (the repositories) at all.
Always use the original one!

Always provide a full debug log. 
I have to guess now that you are missing the widevine binary.
Im sorry, i dont provide instructions for it yet but you can easily get it by installing this kodi service addon:
https://github.com/emilsvennesson/service.widevine

**Comment by DiegoAntonino** [comment]
Thanks for your response (and advices :)).

I am getting 404 when I try to get service.widevine. I also looked all 'emilsvennesson' repositories and I didn't find it. Is 'service.widevine' a public repository? 

Does [installing widevine on Chromium](https://forum.manjaro.org/t/how-to-install-widevine-on-chromium-to-wacth-netflix/4176) do the job? 

Thanks!
Diego

**Comment by JinRonin** [comment]
I think so yes.
All you need is to find and copy the libwidevinecdm.so to .kodi/cdm

**Comment by DiegoAntonino** [comment]
Hi,

I tried but I am still getting the same error.... Is it possible that my current set up is not supported? I installed kodi over rasbian (apt-get install kodi kodi-inputstream-adaptive kodi-inputstream-rtmp ). 

Here you have the kodi logs:

`pi@tv-pi:~/.kodi/temp $ cat kodi.log
19:30:17.927 T:1911214080  NOTICE: special://profile/ is mapped to: special://masterprofile/
19:30:17.928 T:1911214080  NOTICE: -----------------------------------------------------------------------
19:30:17.928 T:1911214080  NOTICE: Starting Kodi (17.4 Git:20170822-d640e9a747-di). Platform: Linux ARM 32-bit
19:30:17.928 T:1911214080  NOTICE: Using Release Kodi x32 build (version for Raspberry Pi)
19:30:17.928 T:1911214080  NOTICE: Kodi compiled Aug 22 2017 by GCC 6.3.0 for Linux ARM 32-bit version 3.18.11 (201227)
19:30:17.928 T:1911214080  NOTICE: Running on Raspbian GNU/Linux 9 (stretch), kernel: Linux ARM 32-bit version 4.9.41-v7+
19:30:17.936 T:1911214080  NOTICE: FFmpeg version/source: ffmpeg-3.1-kodi
19:30:17.936 T:1911214080  NOTICE: Host CPU: ARMv7 Processor rev 4 (v7l), 4 cores available
19:30:17.936 T:1911214080  NOTICE: ARM Features: Neon enabled
19:30:17.936 T:1911214080  NOTICE: special://xbmc/ is mapped to: /usr/share/kodi
19:30:17.936 T:1911214080  NOTICE: special://xbmcbin/ is mapped to: /usr/lib/arm-linux-gnueabihf/kodi
19:30:17.937 T:1911214080  NOTICE: special://xbmcbinaddons/ is mapped to: /usr/lib/arm-linux-gnueabihf/kodi/addons
19:30:17.937 T:1911214080  NOTICE: special://masterprofile/ is mapped to: /home/pi/.kodi/userdata
19:30:17.937 T:1911214080  NOTICE: special://envhome/ is mapped to: /home/pi
19:30:17.937 T:1911214080  NOTICE: special://home/ is mapped to: /home/pi/.kodi
19:30:17.937 T:1911214080  NOTICE: special://temp/ is mapped to: /home/pi/.kodi/temp
19:30:17.937 T:1911214080  NOTICE: special://logpath/ is mapped to: /home/pi/.kodi/temp
19:30:17.937 T:1911214080  NOTICE: The executable running is: /usr/lib/arm-linux-gnueabihf/kodi/kodi_v7.bin
19:30:17.937 T:1911214080  NOTICE: Local hostname: tv-pi
19:30:17.937 T:1911214080  NOTICE: Log File is located: /home/pi/.kodi/temp//kodi.log
19:30:17.937 T:1911214080  NOTICE: -----------------------------------------------------------------------
19:30:18.029 T:1911214080   ERROR: DBus: Error org.freedesktop.DBus.Error.ServiceUnknown - The name org.freedesktop.UPower was not provided by any .service files
19:30:18.096 T:1911214080  NOTICE: load settings...
19:30:18.251 T:1911214080  NOTICE: Found 2 Lists of Devices
19:30:18.251 T:1911214080  NOTICE: Enumerated PI devices:
19:30:18.251 T:1911214080  NOTICE:     Device 1
19:30:18.252 T:1911214080  NOTICE:         m_deviceName      : HDMI
19:30:18.252 T:1911214080  NOTICE:         m_displayName     : HDMI
19:30:18.252 T:1911214080  NOTICE:         m_displayNameExtra:
19:30:18.252 T:1911214080  NOTICE:         m_deviceType      : AE_DEVTYPE_HDMI
19:30:18.252 T:1911214080  NOTICE:         m_channels        : FL,FR
19:30:18.252 T:1911214080  NOTICE:         m_sampleRates     : 8000,11025,16000,22050,24000,32000,44100,48000,88200,96000,176400,192000
19:30:18.252 T:1911214080  NOTICE:         m_dataFormats     : AE_FMT_FLOAT,AE_FMT_S32NE,AE_FMT_S16NE,AE_FMT_S32LE,AE_FMT_S16LE,AE_FMT_FLOATP,AE_FMT_S32NEP,AE_FMT_S16NEP,AE_FMT_RAW
19:30:18.252 T:1911214080  NOTICE:         m_streamTypes     : STREAM_TYPE_AC3,STREAM_TYPE_EAC3,STREAM_TYPE_DTSHD_CORE,STREAM_TYPE_DTS_2048,STREAM_TYPE_DTS_1024,STREAM_TYPE_DTS_512
19:30:18.252 T:1911214080  NOTICE:     Device 2
19:30:18.252 T:1911214080  NOTICE:         m_deviceName      : Analogue
19:30:18.252 T:1911214080  NOTICE:         m_displayName     : Analogue
19:30:18.252 T:1911214080  NOTICE:         m_displayNameExtra:
19:30:18.252 T:1911214080  NOTICE:         m_deviceType      : AE_DEVTYPE_PCM
19:30:18.252 T:1911214080  NOTICE:         m_channels        : FL,FR
19:30:18.252 T:1911214080  NOTICE:         m_sampleRates     : 48000
19:30:18.252 T:1911214080  NOTICE:         m_dataFormats     : AE_FMT_FLOAT,AE_FMT_S32LE,AE_FMT_S16LE,AE_FMT_FLOATP,AE_FMT_S32NEP,AE_FMT_S16NEP
19:30:18.252 T:1911214080  NOTICE:         m_streamTypes     : No passthrough capabilities
19:30:18.253 T:1911214080  NOTICE:     Device 3
19:30:18.253 T:1911214080  NOTICE:         m_deviceName      : Both
19:30:18.253 T:1911214080  NOTICE:         m_displayName     : HDMI and Analogue
19:30:18.253 T:1911214080  NOTICE:         m_displayNameExtra:
19:30:18.253 T:1911214080  NOTICE:         m_deviceType      : AE_DEVTYPE_PCM
19:30:18.253 T:1911214080  NOTICE:         m_channels        : FL,FR
19:30:18.253 T:1911214080  NOTICE:         m_sampleRates     : 48000
19:30:18.253 T:1911214080  NOTICE:         m_dataFormats     : AE_FMT_FLOAT,AE_FMT_S32LE,AE_FMT_S16LE,AE_FMT_FLOATP,AE_FMT_S32NEP,AE_FMT_S16NEP
19:30:18.253 T:1911214080  NOTICE:         m_streamTypes     : No passthrough capabilities
19:30:18.253 T:1911214080  NOTICE: Enumerated ALSA devices:
19:30:18.253 T:1911214080  NOTICE:     Device 1
19:30:18.253 T:1911214080  NOTICE:         m_deviceName      : @
19:30:18.253 T:1911214080  NOTICE:         m_displayName     : Default (bcm2835 ALSA bcm2835 ALSA)
19:30:18.253 T:1911214080  NOTICE:         m_displayNameExtra:
19:30:18.253 T:1911214080  NOTICE:         m_deviceType      : AE_DEVTYPE_PCM
19:30:18.253 T:1911214080  NOTICE:         m_channels        : FL,FR
19:30:18.253 T:1911214080  NOTICE:         m_sampleRates     : 8000,11025,16000,22050,32000,44100,48000
19:30:18.253 T:1911214080  NOTICE:         m_dataFormats     : AE_FMT_S16NE,AE_FMT_S16LE,AE_FMT_U8
19:30:18.253 T:1911214080  NOTICE:         m_streamTypes     : No passthrough capabilities
19:30:18.281 T:1911214080  NOTICE: Loaded settings file from special://xbmc/system/advancedsettings.xml
19:30:18.281 T:1911214080  NOTICE: Contents of special://xbmc/system/advancedsettings.xml are...
                                            <?xml version="1.0" encoding="UTF-8" ?>
                                            <advancedsettings>
                                              <fanartres>720</fanartres>
                                              <imageres>540</imageres>
                                              <video>
                                                <busydialogdelayms>750</busydialogdelayms>
                                              </video>
                                              <samba>
                                                <clienttimeout>30</clienttimeout>
                                              </samba>
                                              <cache>
                                                <memorysize>31457280</memorysize>
                                              </cache>
                                            </advancedsettings>
19:30:18.283 T:1911214080  NOTICE: No settings file to load (special://masterprofile/advancedsettings.xml)
19:30:18.283 T:1911214080  NOTICE: Default Video Player: VideoPlayer
19:30:18.283 T:1911214080  NOTICE: Default Audio Player: paplayer
19:30:18.283 T:1911214080  NOTICE: Disabled debug logging due to GUI setting. Level 0.
19:30:18.283 T:1911214080  NOTICE: Log level changed to "LOG_LEVEL_NORMAL"
19:30:18.290 T:1911214080  NOTICE: CMediaSourceSettings: loading media sources from special://masterprofile/sources.xml
19:30:18.291 T:1911214080  NOTICE: Loading player core factory settings from special://xbmc/system/playercorefactory.xml.
19:30:18.293 T:1911214080  NOTICE: Loaded playercorefactory configuration
19:30:18.293 T:1911214080  NOTICE: Loading player core factory settings from special://masterprofile/playercorefactory.xml.
19:30:18.293 T:1911214080  NOTICE: special://masterprofile/playercorefactory.xml does not exist. Skipping.
19:30:18.322 T:1911214080  NOTICE: Running database version Addons27
19:30:18.549 T:1911214080  NOTICE: ADDONS: Using repository repository.xbmc.org
19:30:18.549 T:1911214080  NOTICE: ADDONS: Using repository repository.xbmchub
19:30:18.549 T:1911214080  NOTICE: ADDONS: Using repository repository.kodinerds
19:30:18.791 T:1911214080  NOTICE: Raspberry PI firmware version: Aug  8 2017 12:05:45
                                            Copyright (c) 2012 Broadcom
                                            version 196063438d7bb0769861c86f948ae653727adb7e (clean) (release)
19:30:18.791 T:1911214080  NOTICE: ARM mem: 752MB GPU mem: 256MB MPG2:0 WVC1:0
19:30:18.791 T:1911214080  NOTICE: cache.memorysize: 30MB libass.cache: 0MB
19:30:18.792 T:1911214080  NOTICE: Config:
                                            arm_freq=1200
                                            audio_pwm_mode=1
                                            config_hdmi_boost=5
                                            core_freq=400
                                            desired_osc_freq=0x36ee80
                                            disable_commandline_tags=2
                                            disable_l2cache=1
                                            disable_splash=1
                                            force_eeprom_read=1
                                            force_pwm_open=1
                                            framebuffer_ignore_alpha=1
                                            framebuffer_swap=1
                                            gpu_freq=300
                                            hdmi_channel_map=8
                                            hdmi_force_cec_address=65535
                                            hdmi_force_hotplug=1
                                            init_uart_clock=0x2dc6c00
                                            lcd_framerate=60
                                            over_voltage_avs=0x186a0
                                            over_voltage_avs_boost=0x16e36
                                            overscan_bottom=48
                                            overscan_left=48
                                            overscan_right=48
                                            overscan_top=48
                                            pause_burst_frames=1
                                            program_serial_random=1
                                            sdram_freq=450
                                            second_boot=1
                                            temp_limit=85
19:30:18.792 T:1911214080  NOTICE: Config:
                                            device_tree=-
19:30:18.792 T:1911214080  NOTICE: InitWindowSystem: Using EGL Implementation: raspberrypi
19:30:18.793 T:1911214080  NOTICE: Found resolution 640 x 480 for display 0 with 640 x 480 @ 60.000000 Hz
19:30:18.793 T:1911214080  NOTICE: Found resolution 640 x 480 for display 0 with 640 x 480 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 800 x 600 for display 0 with 800 x 600 @ 60.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 800 x 600 for display 0 with 800 x 600 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1024 x 768 for display 0 with 1024 x 768 @ 60.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1024 x 768 for display 0 with 1024 x 768 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1280 x 1024 for display 0 with 1280 x 1024 @ 60.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1280 x 1024 for display 0 with 1280 x 1024 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1360 x 768 for display 0 with 1360 x 768 @ 60.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1360 x 768 for display 0 with 1360 x 768 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 720 x 480 for display 0 with 720 x 480 @ 60.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 720 x 480 for display 0 with 720 x 480 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 60.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 60.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 59.940063 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 24.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 23.976025 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 30.000000 Hz
19:30:18.794 T:1911214080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 29.970032 Hz
19:30:18.794 T:1911214080  NOTICE: Found (1920x1080@60.000000) at 30, setting to RES_DESKTOP at 16
19:30:18.794 T:1911214080  NOTICE: Checking resolution 16
19:30:19.136 T:1911214080  NOTICE: GL_VENDOR = Broadcom
19:30:19.136 T:1911214080  NOTICE: GL_RENDERER = VideoCore IV HW
19:30:19.136 T:1911214080  NOTICE: GL_VERSION = OpenGL ES 2.0
19:30:19.136 T:1911214080  NOTICE: GL_SHADING_LANGUAGE_VERSION = OpenGL ES GLSL ES 1.00
19:30:19.136 T:1911214080  NOTICE: GL_EXTENSIONS = GL_OES_compressed_ETC1_RGB8_texture GL_OES_compressed_paletted_texture GL_OES_texture_npot GL_OES_depth24 GL_OES_vertex_half_float GL_OES_EGL_image GL_OES_EGL_image_external GL_EXT_discard_framebuffer GL_OES_rgb8_rgba8 GL_OES_depth32 GL_OES_mapbuffer GL_EXT_texture_format_BGRA8888 GL_APPLE_rgb_422 GL_EXT_debug_marker
19:30:20.329 T:1890579200  NOTICE: Running database version Addons27
19:30:20.333 T:1890579200  NOTICE: Running database version ViewModes6
19:30:20.338 T:1890579200  NOTICE: Running database version Textures13
19:30:20.350 T:1890579200  NOTICE: Running database version MyMusic60
19:30:20.365 T:1890579200  NOTICE: Running database version MyVideos107
19:30:20.370 T:1890579200  NOTICE: Running database version TV29
19:30:20.373 T:1890579200  NOTICE: Running database version Epg11
19:30:20.378 T:1911214080  NOTICE: start dvd mediatype detection
19:30:20.450 T:1770406656  NOTICE: Register - new cec device registered on cec->RPI: CEC Adapter (2708:1001)
19:30:21.016 T:1911214080 WARNING: JSONRPC: Could not parse type "Setting.Details.SettingList"
19:30:21.365 T:1911214080  NOTICE: initialize done
19:30:21.365 T:1911214080  NOTICE: Running the application...
19:30:21.378 T:1911214080  NOTICE: starting zeroconf publishing
19:30:21.378 T:1911214080  NOTICE: starting upnp client
19:30:21.458 T:1614095104  NOTICE: ES: Starting UDP Event server on port 9777
19:30:21.459 T:1614095104  NOTICE: UDP: Listening on port 9777 (ipv6 : false)
19:30:23.170 T:1682907904  NOTICE: Indigo NOTICE: STARTING Indigo SERVICE
19:30:39.322 T:1699685120   ERROR: GetDirectory - Error getting
19:30:40.425 T:1709024000   ERROR: Previous line repeats 6 times.
19:30:40.425 T:1709024000  NOTICE: [plugin.video.dazn] country: ca language: en arguments: 1.0.2
19:30:45.830 T:1709024000  NOTICE: [plugin.video.dazn] country: ca language: en arguments: {'title': ["What's On"], 'params': ['PageType:Home;ContentType:None'], 'id': ['Scheduled'], 'mode': ['rail']}
19:30:56.188 T:1709024000  NOTICE: [plugin.video.dazn] country: ca language: en arguments: {'title': ['Italy v Serbia (Catch Up)'], 'params': ['6bu7i84w77959skcp78n3t2sq'], 'id': ['icegmw4148ro1pfq5l7b99szm'], 'mode': ['play']}
19:30:57.596 T:1911214080  NOTICE: VideoPlayer: Opening: https://dc4livecadazn-a.akamaihd.net/live/channel/1004/all/stream.mpd?event_id=61187624427&h=629fd941bcc5843cff0a6b3370bab4ee
19:30:57.596 T:1911214080 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
19:30:57.602 T:1563763456  NOTICE: Creating InputStream
19:30:58.093 T:1563763456   ERROR: AddOnLog: InputStream Adaptive: Could not open / parse mpdURL (https://dc4livecadazn-a.akamaihd.net/live/channel/1004/all/stream.mpd?event_id=61187624427&h=629fd941bcc5843cff0a6b3370bab4ee)
19:30:58.094 T:1563763456   ERROR: CVideoPlayer::OpenInputStream - error opening [https://dc4livecadazn-a.akamaihd.net/live/channel/1004/all/stream.mpd?event_id=61187624427&h=629fd941bcc5843cff0a6b3370bab4ee]
19:30:58.094 T:1563763456  NOTICE: CVideoPlayer::OnExit()
19:30:58.094 T:1911214080   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.dazn/?id=icegmw4148ro1pfq5l7b99szm&params=6bu7i84w77959skcp78n3t2sq&mode=play&title=Italy+v+Serbia+%28Catch+Up%29]
19:30:58.104 T:1911214080  NOTICE: CVideoPlayer::CloseFile()
19:30:58.105 T:1911214080  NOTICE: VideoPlayer: waiting for threads to exit
19:30:58.105 T:1911214080  NOTICE: VideoPlayer: finished waiting
19:30:58.105 T:1911214080  NOTICE: CVideoPlayer::CloseFile()
19:30:58.105 T:1911214080  NOTICE: VideoPlayer: waiting for threads to exit
19:30:58.105 T:1911214080  NOTICE: VideoPlayer: finished waiting
pi@tv-pi:~/.kodi/temp $
`

**Comment by JinRonin** [comment]
ok try installing the service addon
https://www.dropbox.com/s/f8lppzr8vsxijni/service.widevine-master.zip?dl=1
or this command:
`curl -Ls http://nmacleod.com/public/libreelec/getwidevine.sh | bash

if its not working please enable debug logging and upload the log to pastebin. not here.`

**Comment by DiegoAntonino** [comment]
I got an error saying it does not find  **inputstream.adaptive**... Since I have done many things and i don't know if I cloud have broke something, I've  decided to start from scratch... 

```
1. I removed kodi, kodi-inputstream-adaptive and kodi-inputstream-rtmp
2. I installed them again
3. I try to install Dazn from your repository (as you suggested) and I got the same message as before (_The dependency on inputstream.adaptive version 2.0.10 could not be satisfied_), Here you can find the [debug logs](http://paste.debian.net/986138/)
```

I don't know how kodi manage the add-on but this is the content of /.kodi/addons (where I don't see any inputstream)

```
~/.kodi/addons $ ls -lrt
total 52
drwxr-xr-x 2 pi pi 4096 Sep 14 19:42 metadata.common.musicbrainz.org
drwxr-xr-x 2 pi pi 4096 Sep 14 19:42 metadata.common.fanart.tv
drwxr-xr-x 2 pi pi 4096 Sep 14 19:42 metadata.common.theaudiodb.com
drwxr-xr-x 2 pi pi 4096 Sep 14 19:42 metadata.common.imdb.com
drwxr-xr-x 3 pi pi 4096 Sep 14 19:42 metadata.artists.universal
drwxr-xr-x 2 pi pi 4096 Sep 14 19:42 metadata.common.themoviedb.org
drwxr-xr-x 3 pi pi 4096 Sep 14 19:42 metadata.themoviedb.org
drwxr-xr-x 3 pi pi 4096 Sep 14 19:42 metadata.album.universal
drwxr-xr-x 3 pi pi 4096 Sep 14 19:42 metadata.tvdb.com
drwxr-xr-x 2 pi pi 4096 Sep 14 19:42 packages
drwxr-xr-x 4 pi pi 4096 Sep 14 19:43 service.xbmc.versioncheck
drwxr-xr-x 2 pi pi 4096 Sep 14 19:43 temp
drwxr-xr-x 2 pi pi 4096 Sep 14 19:43 repository.kodinerds
```

So, I found then on these locations:

```
$ sudo find / -name inputstream.adaptive
/usr/lib/arm-linux-gnueabihf/kodi/addons/inputstream.adaptive
/usr/share/kodi/addons/inputstream.adaptive
```

As I mentioned before I have not clue how kodi was designed to work... just trying to give you more info of my current set up (which I believe it's very standard) :)

**Comment by JinRonin** [comment]
Yes the addon is not located in .kodi/addons.
You can check if its installed and enabled when you go to my addons - videoplayer inputstream

If you just want to use kodi on your pi you can try the latest libreelec.tv version (with kodi 17.4) and run the curl command ive posted

**Comment by DiegoAntonino** [comment]
Hi Jin,

Following your advises I made it work!!! :)

That what I did:

1. Installed kodi, kodi-inputstream-adaptive and kodi-inputstream-rtmp (apt-get install .......) 
2. enable addons (go to my addons - videoplayer inputstream)
3. I installed widevine on Chromium using above link (this may not be necessary, but i have done it before)
4. Look where is installed libwidevinecdm.so and libssd_wv.so

```
sudo find / -name libwidevinecdm.so  --> (this may not be necessary)
sudo find / -name libssd_wv.so
```
5. create a symbolic link in ~/.kodi/cdm/
```
ln -fs /usr/lib/chromium-browser/libwidevinecdm.so  ~/.kodi/cdm/libwidevinecdm.so --> (this may not be necessary)
ln -fs /usr/lib/chromium-browser/libssd_wv.so  ~/.kodi/cdm/libssd_wv.so
```

Now I am enjoin Dazn :)

Thanks for your great support!
Have a good weekend!
Diego
